lib/elixir/test/elixir/string_test.exs

```diff
@@ -274,6 +274,8 @@ defmodule StringTest do
 
     assert String.replace("a,b,c", ~r/,(.)/, ",\\1\\1") == "a,bb,cc"
     assert String.replace("a,b,c", ~r/,(.)/, ",\\1\\1", global: false) == "a,bb,c"
+    assert String.replace("a.b.c", ~r/\./, %{"." => "-"}) == "a-b-c"
+    assert String.replace("abba", ~r/a|b/, %{"a" => "1", "b" => "2"}) == "1221"
```

Comment:
We have no tests if the second argument is a string and I think it wouldn't work with the given implementation. It would actually be hard to implement it for strings as second argument efficiently so I would maybe restrict this to the Regex only. See how Regex.replace is implemented with function replacements for ideas. :)